### ISSUE TITLE
Massive FPS drops with "Galaxy map starfields rendering" on medium zoom levels

### ISSUE THREAD

**Issue by Morlic-fo** [freeorion/freeorion#1403]
### Environment
* **FreeOrion Version**: build 2017-03-14.853af0e
* **Operating System**: Windows 7
* **Graphic card used**: Nvidia GT520M
* **Fetched as** Weekly test build

### Description
Playing on my laptop I experience massive FPS drops when the option "Galaxy map starfield rendering" is active and the galaxy map is on a "medium" zoom level (see screenshots).

The FPS I get are (see screenshots for test cases)
Maximally zoomed out:   ~120 off --> ~90 on
Medium zoom level: ~120 off --> 15-20 on
Maximally zoomed in: both ~120

The FPS drops are so massive that the mouse pointer is lagging and it becomes an annoyance to play.
The "default" zoom level after fresh install and game start is in the critical FPS drop range for me.

### Steps to reproduce:
1. Start a new game
2. For different zoom levels, try turning on and off the Galaxy map starfield rendering setting under 
Options > Galaxy Map.

### Expected result:
No FPS drops would obviously be great.

I understand that stuff like this can also be a graphics driver issue or simply related to the low-spec setup. Still, I wonder if it would be a good idea to default the setting to off  if other people can reproduce.

![faroff](https://cloud.githubusercontent.com/assets/11706951/23971606/ecf9ab7a-09ce-11e7-9328-7a31032374cb.png)
![faron](https://cloud.githubusercontent.com/assets/11706951/23971607/ecfbd0bc-09ce-11e7-8aa5-309013f4be89.png)
![mediumon](https://cloud.githubusercontent.com/assets/11706951/23971608/ecfc1bd0-09ce-11e7-97b7-ca86db942234.png)
![mediumoff](https://cloud.githubusercontent.com/assets/11706951/23971610/ecff9940-09ce-11e7-9cc0-946a46ec77ee.png)
![closeoff](https://cloud.githubusercontent.com/assets/11706951/23971609/ecfcd8b8-09ce-11e7-95d3-1d177ebf2dfa.png)
![closeon](https://cloud.githubusercontent.com/assets/11706951/23971611/ed0218fa-09ce-11e7-9668-749ea4567496.png)

**Comment by adrianbroher** [comment]
> Still, I wonder if it would be a good idea to default the setting to off if other people can reproduce.

No, why should we cater for low end systems by default?

Also in general the rendering in FO is horrible by default.  Past experience showed that the render code emits much more draw calls than needed and this just destroys the performance.

**Comment by geoffthemedio** [comment]
https://github.com/freeorion/freeorion/blob/master/UI/MapWnd.cpp#L1737

The starfields are rendered with a single glDrawArrays call.

**Comment by Morlic-fo** [comment]
> No, why should we cater for low end systems by default?

First impressions are important. If I was playing this game for the first time, I would have quit FO for good after 2 minutes because of the lag. The option to turn off is not immediately found either as it is well hidden among other options in the Galaxy Map options (where I would not look as new player). A quick-to-find Graphics-settings which would turn it off when set to Low would also be a good improvement.

You are of course right that it should not default to off if it affects only a small percentage of potential users. If it would affect 50% of potential users, it would be quite significant and should be considered a reason to turn it off by default.

Now I can not tell what is the "low spec cutoff" on my own. On my desktop, it is fine. But 95% of players will probably not have those specs either. I don't have endless setups to test what kind of specs are required for the FPS drops to occur or not occur and therefore can not tell how many potential people this will affect.

I also do not have an idea of what the setup of an average Freeorion player is but considering the target group of MoO veterans, I would expect that most are not playing on high end dev/gaming setups.

**Comment by geoffthemedio** [comment]
I don't think this is a 'low spec' isuue, but rather a bad drivers issue. The starfields are not complicated to render, and the problem is only appearing in specific circumstances (certain zoom levels), between which the rendering code does almost nothing different.

Can you double check that your system is actually rendering using the hardware and not a software renderer? My guess is that at particular zoom levels it's getting slowed down by rendering lots of transparent pixels...

**Comment by Vezzra** [comment]
I agree with @geoffthemedio here, doesn't look like a "low spec" issue - after all, in all but what seems a narrow range of zoom levels @Morlic-fo is getting pretty high FPS (in the 90-120 range). Only within that narrow range the FPS suddenly drops to 17.

Just checked on my system, I don't have the problem (FPS are stable throughout all zoom levels), but my machine most definitely is lower spec than @Morlic-fo's. I only get half his FPS, 57-59.

**Comment by geoffthemedio** [comment]
> I only get half his FPS, 57-59.

Might be due to not having adjusted the FPS limit above the default 60.

**Comment by Vezzra** [comment]
Well, oops. However, as I just noticed, the limit FPS feature apparently doesn't work at all on my system. Regardless what I put into the max FPS box has no impact at all on my FPS. They hover barely below 60.

So I guess these 57-59 FPS are what my hardware is actually capable of (my laptop is 6 years old now after all).

**Comment by geoffthemedio** [comment]
For me, you have to switch focus to another window and back for the change to the FPS limit to kick in. Also, turn off gas and stars rendering for a probable FPS boost.

**Comment by Vezzra** [comment]
Ah yes, switching focus worked. Although I thought I even quit and restarted before, but probably did something wrong. However, I can only get *lower* FPS when reducing the max to e.g. 30, but I don't get over 60 when raising the max to e.g. 80.

So I guess the 60 are actually the limit my hardware can manage.

**Comment by geoffthemedio** [comment]
There could be some other frame rate limit in place as well, like to the monitor refresh rate.

There should probly be an issue about the FPS limit not updating without a window focus change.
